libs/datasets/timeseries.py

```diff
@@ -113,14 +107,20 @@ def yield_records(self) -> Iterable[dict]:
 
     def get_subset(self, after=None, columns=tuple()):
         rows_key = dataset_utils.make_rows_key(self.data, after=after,)
-        columns_key = list(columns) if columns else slice(None, None, None)
+        if columns:
+            assert CommonFields.LOCATION_ID in columns
```

Comment:
I think if we have these asserts in the input data itself it makes it easier to reason about it elsewhere.  Why did you move it from the post init to here?

Reply:
I bumped into a bug where LOCATION_ID wasn't in columns but you are correct that the check makes more sense in `__post_init__` so moved them there.